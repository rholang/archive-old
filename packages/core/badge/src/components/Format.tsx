import React, { FC } from 'react';

const { Fragment } = React;

interface Props {
  /** The number to format. */
  children?: number | string;
  /** The maximum value to display. If value is 100, and max is 50, "50+" will be displayed */
  max?: number;
}

const Format: FC<Props> = props => {
  let formatted: number | string = '';
  let { children = 0, max = 0 } = props;

  if (children < 0) {
    children = 0;
  }

  if (max < 0) {
    max = 0;
  }

  if (max && max < children) {
    formatted = `${max}+`;
  } else if (children === Infinity) {
    formatted = 'âˆž';
  } else {
    formatted = children;
  }

  return <Fragment>{formatted}</Fragment>;
};

export default Format;
