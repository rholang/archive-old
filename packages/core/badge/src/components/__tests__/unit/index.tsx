import { mount } from 'enzyme';
import React from 'react';
import Badge from '../..';
import Format from '../../Format';

function getFormatProps(badge: React.ReactElement<any>) {
  return mount(badge)
    .find(Format)
    .props();
}

test('snapshot', () => {
  expect(mount(<Badge />)).toMatchSnapshot();
});

test('children', () => {
  expect(getFormatProps(<Badge>{100}</Badge>)).toMatchObject({ children: 100 });
});

test('using a string', () => {
  expect(mount(<Badge>+100</Badge>).props()).toMatchObject({
    children: '+100',
  });
});
