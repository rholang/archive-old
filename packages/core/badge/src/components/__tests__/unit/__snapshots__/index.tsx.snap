// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  background-color: #DFE1E6;
  color: #172B4D;
  border-radius: 2em;
  display: inline-block;
  font-size: 12px;
  font-weight: normal;
  line-height: 1;
  min-width: 1px;
  padding: 0.16666666666667em 0.5em;
  text-align: center;
}

<Badge>
  <Provider>
    <Consumer>
      <Consumer
        appearance="default"
        mode="light"
      >
        <styled.span
          backgroundColor="#DFE1E6"
          textColor="#172B4D"
        >
          <span
            className="c0"
          >
            <Format
              max={99}
            >
              0
            </Format>
          </span>
        </styled.span>
      </Consumer>
    </Consumer>
  </Provider>
</Badge>
`;
